### ISSUE TITLE
Python 3.13a4: ``_Py_atomic_load_ssize_relaxed``: identifier not found

### ISSUE THREAD

**Issue by wjakob** [python/cpython#116078]
# Bug report

### Bug description:

Dear CPython team,

I bumped the nanobind test suite build matrix to include 3.13a4. This works except for compilation on windows via MSVC, which fails with an error complaining that ``_Py_atomic_load_ssize_relaxed`` is not defined. 

```
C:\hostedtoolcache\windows\Python\3.13.0-alpha.4\x64\include\object.h(433,5): error C3861: '_Py_atomic_store_ssize_relaxed': identifier not found
```

Link: https://github.com/wjakob/nanobind/actions/runs/8087892173/job/22100798452

I think that including `Python.h` from an extension is all that's needed to trigger this. I know that there has been a lot of activity in this part of the code related to free-threading. But I am not creating a free-threading build.

### CPython versions tested on:

3.13, CPython main branch

### Operating systems tested on:

Windows

**Comment by zooba** [comment]
Unfortunately, due to a build issue in 3.13a4, you may still have the `Py_GIL_DISABLED` flag set (because it's hardcoded into the header, and the wrong header shipped - see #115582). That's likely the cause of this issue.

If you're able to, could you patch that definition (in `pyconfig.h`) to 0 to see if that fixes the issue? I've been building plenty of my own stuff with 3.13a4, though with a fixed build, and have not seen this particular problem.

**Comment by wjakob** [comment]
I'm using the [GitHub setup-python action](https://github.com/actions/setup-python) and so have no influence over the Python package that is installed.
